src/emqttd.erl

```diff
@@ -45,86 +30,66 @@
+env(Group, Name) -> proplists:get_value(Name, env(Group)).
+
+%% @doc Start Listeners of the broker.
 -spec start_listeners() -> any().
-start_listeners() ->
-    {ok, Listeners} = application:get_env(?APP, listeners),
-    lists:foreach(fun start_listener/1, Listeners).
+start_listeners() -> lists:foreach(fun start_listener/1, env(listeners)).
 
 %% Start mqtt listener
 -spec start_listener(listener()) -> any().
-start_listener({mqtt, Port, Options}) ->
-    start_listener(mqtt, Port, Options);
+start_listener({mqtt, Port, Opts}) -> start_listener(mqtt, Port, Opts);
 
 %% Start mqtt(SSL) listener
-start_listener({mqtts, Port, Options}) ->
-    start_listener(mqtts, Port, Options);
+start_listener({mqtts, Port, Opts}) -> start_listener(mqtts, Port, Opts);
 
 %% Start http listener
-start_listener({http, Port, Options}) ->
-    MFArgs = {emqttd_http, handle_request, []},
-    mochiweb:start_http(Port, Options, MFArgs);
+start_listener({http, Port, Opts}) ->
+    mochiweb:start_http(Port, Opts, {emqttd_http, handle_request, []});
 
 %% Start https listener
-start_listener({https, Port, Options}) ->
-    MFArgs = {emqttd_http, handle_request, []},
-    mochiweb:start_http(Port, Options, MFArgs).
+start_listener({https, Port, Opts}) ->
+    mochiweb:start_http(Port, Opts, {emqttd_http, handle_request, []}).
 
-start_listener(Protocol, Port, Options) ->
+start_listener(Protocol, Port, Opts) ->
     MFArgs = {emqttd_client, start_link, [env(mqtt)]},
-    esockd:open(Protocol, Port, merge_sockopts(Options) , MFArgs).
+    esockd:open(Protocol, Port, merge_sockopts(Opts), MFArgs).
 
 merge_sockopts(Options) ->
     SockOpts = emqttd_opts:merge(?MQTT_SOCKOPTS,
                                  proplists:get_value(sockopts, Options, [])),
     emqttd_opts:merge(Options, [{sockopts, SockOpts}]).
 
-%%------------------------------------------------------------------------------
 %% @doc Stop Listeners
-%% @end
-%%------------------------------------------------------------------------------
-stop_listeners() ->
-    {ok, Listeners} = application:get_env(?APP, listeners),
-    lists:foreach(fun stop_listener/1, Listeners).
+stop_listeners() -> lists:foreach(fun stop_listener/1, env(listeners)).
 
-stop_listener({Protocol, Port, _Options}) ->
-    esockd:close({Protocol, Port}).
+stop_listener({Protocol, Port, _Opts}) -> esockd:close({Protocol, Port}).
 
+%% @doc load all modules
 load_all_mods() ->
     lists:foreach(fun load_mod/1, env(modules)).
 
 load_mod({Name, Opts}) ->
     Mod = list_to_atom("emqttd_mod_" ++ atom_to_list(Name)),
     case catch Mod:load(Opts) of
-        {ok, _State}     -> lager:info("load module ~s successfully", [Name]);
-        {'EXIT', Reason} -> lager:error("load module ~s error: ~p", [Name, Reason])
+        ok               -> lager:info("Load module ~s successfully", [Name]);
+        {error, Error}   -> lager:error("Load module ~s error: ~p", [Name, Error]);
```

Comment:
Line 86 is too long:         {error, Error}   -> lager:error("Load module ~s error: ~p", [Name, Error]);.